File: 2021/01.py
puzzle = [int(i) for i in open("01.txt").readlines()]

# Part 1
count = 0
for i in range(1, len(puzzle)):
    if puzzle[i] > puzzle[i - 1]:
        count += 1

print(count)

# Part 2
count2 = 0
for i in range(1, len(puzzle)-2):
    if sum(puzzle[i:i+3]) > sum(puzzle[i-1:i+2]):
        count2 += 1

print(count2)

File: 2022/01.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"sort"
	"strconv"
)

func sum(array []int) int {
	result := 0
	for _, item := range array {
		result += item
	}
	return result
}

func count() []int {
	file, err := os.Open("01.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var calories []int
	var sum int
	for scanner.Scan() {
		l := scanner.Text()
		if l == "" {
			calories = append(calories, sum)
			sum = 0
		} else {
			i, err := strconv.Atoi(l)
			if err != nil {
				log.Fatal(err)
			}
			sum += i
		}
	}
	sort.Ints(calories)
	return calories

}

func main() {
	calories := count()
	fmt.Println(calories[len(calories)-1])
	fmt.Println(sum(calories[len(calories)-3:]))
}
